app/src/main/java/us/shandian/giga/ui/fragment/MissionsFragment.java

```diff
@@ -249,6 +260,23 @@ private void recoverMission(@NonNull DownloadMission mission) {
         );
     }
 
+    public int getMaxActionButtons() {
+        Configuration configuration = mContext.getResources().getConfiguration();
+        int widthDp = configuration.screenWidthDp;
+        int heightDp = configuration.screenHeightDp;
+        int smallest = configuration.smallestScreenWidthDp;
+
+        if (smallest > 600 || widthDp > 600) {
+            return 5;
```

Comment:
I think this is quite hard to say exactly, how much space we will have. When using the English version, this might be correct, but I guess when using a different locale, this can be different.